app/controllers/concerns/part_of_claim_journey.rb

```diff
@@ -11,12 +11,23 @@ module PartOfClaimJourney
   private
 
   def check_whether_closed_for_submissions
-    unless journey_configuration.open_for_submissions?
+    unless journey.open_for_submissions?(access_code)
       @availability_message = journey_configuration.availability_message
       render "static_pages/closed_for_submissions", status: :service_unavailable
     end
   end
 
+  def access_code
+    if journey_session
+      journey_session.answers.service_access_code
+    else
+      # We've been redirected from the landing page and this callback
+      # is running before the new action, so we're yet to create the
+      # journey session.
+      params.fetch(:answers, {})[:service_access_code]
+    end
+  end
```

Comment:
Ideally we could move the check_whether_closed_for_submission to run after the journey session is created so we don't need this branch and can assume there's always a `journey_session`, however there's some trickiness here due to callbacks depending on each other. On an unrelated note we should probably remove this module and inline it's functionality in the `ClaimsController` as only the `set_cache_headers` method is shared (and that can go in the base public controller).